Sources/Subs.php

```diff
@@ -6855,19 +6855,16 @@ function getUserTimezone($id_member = null)
 	}
 
 	// Look it up in the database.
-	if (isset($id_member))
-	{
-		$request = $smcFunc['db_query']('', '
-			SELECT timezone
-			FROM {db_prefix}members
-			WHERE id_member = {int:id_member}',
-			array(
-				'id_member' => $id_member,
-			)
-		);
-		list($timezone) = $smcFunc['db_fetch_row']($request);
-		$smcFunc['db_free_result']($request);
-	}
+	$request = $smcFunc['db_query']('', '
+		SELECT timezone
+		FROM {db_prefix}members
+		WHERE id_member = {int:id_member}',
+		array(
+			'id_member' => $id_member,
+		)
+	);
+	list($timezone) = $smcFunc['db_fetch_row']($request);
+	$smcFunc['db_free_result']($request);
```

Comment:
You should put back the if logic but use !empty.  We can't query the database for id_member = 0.
Unless I missed something.

Reply:
You are correct. Running the query doesn't break anything, but it is pointless. I have restored it in #7189.